src/include/catalog/database_catalog.h

```diff
@@ -36,490 +27,233 @@ class Index;
-   * obtain the authoritative schema for this table.
-   */
-  bool UpdateSchema(common::ManagedPointer<transaction::TransactionContext> txn, table_oid_t table, Schema *new_schema);
-
-  /**
-   * Get the visible schema describing the table.
-   * @param txn for the operation
-   * @param table corresponding to the requested schema
-   * @return the visible schema object for the identified table
-   */
+  /** @brief Get the schema for the specified table. */
   const Schema &GetSchema(common::ManagedPointer<transaction::TransactionContext> txn, table_oid_t table);
+  /** @brief Get the index schema for the specified index. */
+  const IndexSchema &GetIndexSchema(common::ManagedPointer<transaction::TransactionContext> txn, index_oid_t index);
 
   /**
-   * A list of all constraints on this table
-   * @param txn for the operation
-   * @param table being queried
-   * @return vector of OIDs for all of the constraints that apply to this table
+   * @brief Rename a table.
+   *
+   * @param txn         The transaction to rename the table in.
+   * @param table       The table to be renamed.
+   * @param name        The new name for the table.
+   * @return            True if the rename succeeded. False otherwise.
+   *
+   * TODO(WAN): if this logic can be pushed to PgCoreImpl, update this comment to match style
    */
-  std::vector<constraint_oid_t> GetConstraints(common::ManagedPointer<transaction::TransactionContext> txn,
-                                               table_oid_t table);
+  bool RenameTable(common::ManagedPointer<transaction::TransactionContext> txn, table_oid_t table,
+                   const std::string &name);
 
   /**
-   * A list of all indexes on the given table
-   * @param txn for the operation
-   * @param table being queried
-   * @return vector of OIDs for all of the indexes on this table
+   * @brief Update the schema of the table.
+   *
+   * Apply a new schema to the given table.
+   * The changes will modify the latest schema as provided by the catalog.
+   * There is no guarantee that the OIDs for modified columns will be stable across a schema change.
+   *
+   * @param txn         The transaction to update the table's schema in.
+   * @param table       The table whose schema should be updated.
+   * @param new_schema  The new schema to update the table to.
+   * @return            True if the update succeeded. False otherwise.
+   *
+   * @warning           The catalog accessor assumes it takes ownership of the schema object that is passed.
+   *                    As such, there is no guarantee that the pointer is still valid when this function returns.
+   *                    If the caller needs to reference the schema object after this call, the caller should use
+   *                    the GetSchema function to obtain the authoritative schema for this table.
+   *
+   * TODO(WAN): if this logic can be pushed to PgCoreImpl, update this comment to match style
```

Comment:
Unlike the other function with this comment, I think this one ultimately has to stay at the `DatabaseCatalog`-level because it may have wider impacts then just `Core` (e.g., updating `Constraints` and possibly other tables).

Reply:
Removed TODO.